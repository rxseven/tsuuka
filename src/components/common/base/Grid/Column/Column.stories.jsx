import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Column from './index';

const { content } = mock.elements;
const title = 'Components/Common/Base/Grid/Column';
const component = <Column>{content}</Column>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
