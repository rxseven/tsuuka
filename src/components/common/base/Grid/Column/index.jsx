import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  size: PropTypes.string
});

const defaultProps = {
  size: 'col'
};

function Column({ children, size }) {
  return <div className={size}>{children}</div>;
}

Column.propTypes = propTypes;
Column.defaultProps = defaultProps;

export default Column;
