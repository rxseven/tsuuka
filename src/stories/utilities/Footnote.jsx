import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

function Footnote({ children }) {
  return <Frame>{children}</Frame>;
}

export default Footnote;
