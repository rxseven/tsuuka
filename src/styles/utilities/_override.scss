// Reset button's default styles
button:focus {
  box-shadow: none !important;
  outline: 0 !important;
}
