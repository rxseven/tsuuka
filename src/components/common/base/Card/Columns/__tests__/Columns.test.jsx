import React from 'react';

import { factory, toMarkup } from 'tests/utilities';
import Column from '../index';

// Arrange
const seed = { content: 'content' };
const source = { children: <span>{seed.content}</span> };
const input = { ...seed, ...source };

// Setup
function setup(props) {
  return factory(Column, source, props);
}

// Test suites
describe('<Card.Column />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render passed children correctly', () => {
    const expected = {
      content: input.content,
      html: input.children
    };
    const { component } = setup();

    expect(component).toHaveTextContent(expected.content);
    expect(component).toContainHTML(toMarkup(expected.html));
  });
});
