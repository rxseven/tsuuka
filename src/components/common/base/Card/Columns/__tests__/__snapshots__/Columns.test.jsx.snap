// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card.Column /> Snapshot tests should render correctly 1`] = `
<div
  className="card-columns"
>
  content
</div>
`;
